src/main/java/com/google/api/pathtemplate/PathTemplate.java

```diff
@@ -627,37 +628,41 @@ private boolean match(
           }
           if (currentVar != null) {
             // Create or extend current match
-            String current = values.get(currentVar);
-            if (current == null) {
-              values.put(currentVar, next);
-            } else {
-              values.put(currentVar, current + "/" + next);
-            }
+            values.put(currentVar, concatCaptures(values.get(currentVar), next));
           }
-          if (seg.kind() == SegmentKind.PATH_WILDCARD) {
-            // Compute the number of additional input the ** can consume. This
-            // is possible because we restrict patterns to have only one **.
-            int segsToMatch = 0;
-            for (int i = segPos; i < segments.size(); i++) {
-              switch (segments.get(i).kind()) {
-                case BINDING:
-                case END_BINDING:
-                  // skip
-                  continue;
-                default:
-                  segsToMatch++;
-              }
-            }
-            int available = (input.size() - inPos) - segsToMatch;
-            while (available-- > 0) {
-              values.put(currentVar, values.get(currentVar) + "/" + decodeUrl(input.get(inPos++)));
+          break;
+        case PATH_WILDCARD:
+          // Compute the number of additional input the ** can consume. This
+          // is possible because we restrict patterns to have only one **.
```

Comment:
Where do we restrict patterns to have a single `**`?

Reply:
https://github.com/googleapis/api-common-java/blob/master/src/main/java/com/google/api/pathtemplate/PathTemplate.java#L925